client/src/App.tsx

```diff
@@ -29,6 +29,19 @@ const App = () => {
     setNotes(newNotes);
   };
 
+  useEffect(() => {
+    const savedNotes = JSON.parse(
+      localStorage.getItem("react-notes-app-data")!
```

Comment:
what if `localStorage.getItem("react-notes-app-data")` returns undefined? `JSON.parse(undefined)` will throw an exception.

I suggest to wrap `JSON.parse()` inside try catch.

Reply:
`localStorage.getItem("react-notes-app-data")` returns `null` in case value is not present in local storage, and `JSON.parse(null)` doesn't throw an exception at all. So we are fine by this for now 👍🏼